spec/Section 3 -- Type System.md

```diff
@@ -2164,3 +2173,115 @@ to the relevant IETF specification.
 ```graphql example
 scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
 ```
+
+### @defer
+
+```graphql
+directive @defer(
+  if: Boolean! = true
+  label: String
+) on FRAGMENT_SPREAD | INLINE_FRAGMENT
+```
+
+The `@defer` directive may be provided on a fragment spread or inline fragment
+to indicate that execution of the related selection set should be deferred. When
+a request includes the `@defer` directive, it may return an _incremental stream_
+consisting of an _initial response_ containing all non-deferred data, followed
+by one or more _subsequent payloads_ including the deferred data.
+
+The `@include` and `@skip` directives take precedence over `@defer`.
+
+```graphql example
+query myQuery($shouldDefer: Boolean! = true) {
+  user {
+    name
+    ...someFragment @defer(label: "someLabel", if: $shouldDefer)
+  }
+}
+fragment someFragment on User {
+  id
+  profile_picture {
+    uri
+  }
+}
+```
+
+#### @defer Arguments
+
+- `if: Boolean! = true` - When `true`, fragment _should_ be deferred (see
+  related note below). When `false`, fragment must not be deferred. Defaults to
+  `true` when omitted.
+- `label: String` - An optional string literal (variables are disallowed) used
+  by GraphQL clients to identify data in the _incremental stream_ and associate
+  it with the corresponding defer directive. If provided, the GraphQL service
+  must include this label in the corresponding pending object within the
+  _incremental stream_. The `label` argument must be unique across all `@defer`
+  and `@stream` directives in the document.
+
+### @stream
+
+```graphql
+directive @stream(
+  initialCount: Int! = 0
+  if: Boolean! = true
+  label: String
+) on FIELD
+```
+
+The `@stream` directive may be provided for a field whose type incorporates a
+`List` type modifier. The directive enables returning a partial list initially,
+followed by additional items in subsequent payloads. Should the field type
+incorporate multiple `List` type modifiers, only the outermost list is streamed.
```

Comment:
```suggestion
followed by additional items in subsequent payloads. If the field type
incorporates multiple `List` type modifiers, only the outermost list is streamed.
```

Extremely minor rephrasing that I think may be more "spec sounding". Feel free to ignore if that doesn't make sense.